### PR TITLE
Feat/dice loss averaging

### PR DESCRIPTION
**Pull request by ViriAldi**
#### Changes
Added optional argument `average` to Dice Loss module and function. The current version of the Dice Loss performs only "micro" averaging for multi-class scenario. However, the documentation states that loss is computed for each individual class. `average` by default takes `"micro"` value and leaves the the current behaviour unchanged. I added another option `"macro"` that calculates Dice Loss for each class separately and then averages between classes and samples. The motivation is to add the possibility to handle imbalanced class data (e.g. segmentation of small objects compared to the background). Also added relevant testcases and updated documentation. See [link](https://torchmetrics.readthedocs.io/en/stable/classification/dice.html#functional-interface) for details

Fixes # (issue)

#### Type of change

- [x] 📚  Documentation Update
- [x] 🧪 Tests Cases
- [x] 🔬 New feature (non-breaking change which adds functionality)
- [x] 📝 This change requires a documentation update

#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

File: kornia/losses/dice.py [hunk 1]
```diff
@@ -115,9 +138,10 @@ class DiceLoss(nn.Module):
         >>> output.backward()
     """
 
-    def __init__(self, eps: float = 1e-8) -> None:
+    def __init__(self, average: str | None = "micro", eps: float | None = 1e-8) -> None:
         super().__init__()
+        self.average: str = average
```
Comment (johnnv1) [inline kornia/losses/dice.py:143]:
```suggestion
        self.average = average
```

File: kornia/losses/dice.py [hunk 2]
```diff
@@ -115,9 +138,10 @@ class DiceLoss(nn.Module):
...
         self.eps: float = eps
```
Comment (johnnv1) [inline kornia/losses/dice.py:144]:
```suggestion
        self.eps = eps
```

File: kornia/losses/dice.py [hunk 3]
```diff
@@ -60,20 +67,32 @@ def dice_loss(input: torch.Tensor, target: torch.Tensor, eps: float = 1e-8) -> t
...
+        raise ValueError(f"The `average` has to be one of {possible_average}. Got: {average}")
+
     # compute softmax over the classes axis
     input_soft: torch.Tensor = F.softmax(input, dim=1)
 
     # create the labels one hot tensor
     target_one_hot: torch.Tensor = one_hot(target, num_classes=input.shape[1], device=input.device, dtype=input.dtype)
 
+    # set dimensions for the appropriate averaging
+    dims = (2, 3)
```
Comment (johnnv1) [inline kornia/losses/dice.py:81]: i think this will satisfy mypy (needs to check)
```suggestion
    dims: tuple[int, ...] = (2, 3)
```

File: kornia/losses/dice.py [hunk 4]
```diff
@@ -55,7 +55,7 @@ def dice_loss(
         >>> output = dice_loss(input, target)
         >>> output.backward()
     """
-    if not isinstance(input, torch.Tensor):
+    if not isinstance(input, Tensor):
```
Comment (edgarriba) [inline kornia/losses/dice.py:58]: Ideally we should use KORNIA_CHECK_IS_TENSOR

File: kornia/losses/dice.py [hunk 5]
```diff
@@ -67,18 +67,17 @@ def dice_loss(
...
 
-    possible_average = ("micro", "macro")
-    if average not in possible_average:
-        raise ValueError(f"The `average` has to be one of {possible_average}. Got: {average}")
+    possible_average = {"micro", "macro"}
+    KORNIA_CHECK(average in possible_average, f"The `average` has to be one of {possible_average}. Got: {average}")
 
     # compute softmax over the classes axis
-    input_soft: torch.Tensor = F.softmax(input, dim=1)
+    input_soft: Tensor = F.softmax(input, dim=1)
```
Comment (edgarriba) [inline kornia/losses/dice.py:74]:
```suggestion
    input_soft: Tensor = input.softmax( dim=1)
```

File: kornia/losses/dice.py [hunk 6]
```diff
@@ -48,8 +54,7 @@ def dice_loss(input: torch.Tensor, target: torch.Tensor, eps: float = 1e-8) -> t
         >>> output = dice_loss(input, target)
         >>> output.backward()
     """
-    if not isinstance(input, torch.Tensor):
-        raise TypeError(f"Input type is not a torch.Tensor. Got {type(input)}")
+    KORNIA_CHECK_IS_TENSOR(input, f"Input type is not a torch.Tensor. Got {type(input)}")
```
Comment (johnnv1) [inline kornia/losses/dice.py:57]: we already have this msg on kornia check 😄 
```suggestion
    KORNIA_CHECK_IS_TENSOR(input)
```

**Review by johnnv1 — APPROVED** [review]
